### PR TITLE
[DOC] Add doc for maybe_wrap_dim

### PR DESCRIPTION
**Pull request by tugsbayasgalan**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #63161

Differential Revision: [D30629451](https://our.internmc.facebook.com/intern/diff/D30629451)

### CODE REVIEW

File: aten/src/ATen/WrapDimUtils.h [hunk 1]
```diff
@@ -7,6 +7,10 @@
 namespace at {
 
 static inline int64_t maybe_wrap_dim(int64_t dim, int64_t dim_post_expr, bool wrap_scalar=true) {
+  // dim_post_expr can be negative only when wrap_scalar is true, which will
+  // automatically set it to 1.
```
Comment (zou3519) [inline aten/src/ATen/WrapDimUtils.h:11]: Err I don't know what it means for `dim_post_expr` to be negative. That just sounds like a bug in the code.

I think what we want to say is "if dim_post_expr is 0 and wrap_scalar, then dim must be in the range [-1, 0]. This is a weird special case for scalar tensors and manifests in e.g. torch.sum(scalar_tensor, 0)"

File: aten/src/ATen/WrapDimUtils.h [hunk 2]
```diff
@@ -7,6 +7,10 @@
...
+  // dim should be in the range [-dim_post_expr, dim_post_expr-1]
+  // maximum interval for dim is [-1, 0]
```
[the remark this answers is not in the document]
Comment (tugsbayasgalan) [reply aten/src/ATen/WrapDimUtils.h:13]: yes

[the remark this answers is not in the document]
Comment (tugsbayasgalan) [reply aten/src/ATen/WrapDimUtils.h:13]: I will update.

**Review by zou3519 — APPROVED** [review]
awesome, thank you!

**Comment by tugsbayasgalan** [comment]
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30629451).
